src/node.rs

```diff
@@ -63,7 +63,8 @@ pub struct SawtoothRaftNode<S: StorageExt> {
     service: Box<Service>,
     leader_state: Option<LeaderState>,
     follower_state: Option<FollowerState>,
-    valid_block_backlog: VecDeque<BlockId>,
+    new_block_backlog: VecDeque<Block>,
```

Comment:
what's the difference between the new_block_backlog and the commit_block_backlog?

Reply:
Can you please clarify how these queues or the general approach relates to the raft library's order enforcement mechanisms e.g. entry index?

Reply:
Raft's order enforcement does not guarantee anything about the application itself (i.e. Sawtooth). From the library's perspective, it accepts a state transition (new entry) and saves it to stable storage; it then tells the application to submit that entry. However, the application may not be ready to make that transition--in this case it may not be ready to commit the block because the validator is still in the process of validating said block. So the library makes no guarantees about the application, except that it will only tell the application to commit the entries sequentially. We can rely on the library to send commits in order, but we cannot necessarily rely on the commit messages to arrive in order due to factors that are out of the library's control (like networking delays).